.github/workflows/deploy.yml

```diff
@@ -1,16 +1,27 @@
-name: Deploy to dev
+name: Deploy
 
 on:
+  workflow_call: # Called after tests pass on master branch push
+    inputs:
+      environment:
+        description: 'Environment'
+        type: string # environment type not supported here, yet
+        default: dev
   workflow_dispatch:
-  push:
-    branches: [master]
+    inputs:
+      environment:
+        description: 'Environment'
```

Comment:
Maybe `'Environment (dev or prod)'` so hint the user what to type here? Otherwise there might be a bit of hesitancy about whether it's `dev` or `development`, for example.